src/audio.rs

```diff
@@ -276,13 +278,19 @@ fn extr_neighbor<F>(v :&Vec<u32>, x :usize,
 	// If "a bigger b" means compare(b, a) == true
 	let split_idx = extr_idx;
 	let smm = sm.split_at(split_idx).1;
-	for (idx, val) in smm.iter().cloned().enumerate() {
-		if compare(val, bound) && compare(max_val, val) {
-			extr_idx = idx + split_idx;
-			max_val = val;
-		}
-	}
-	return (extr_idx, max_val);
+	let max_val = smm.into_iter()
```

Comment:
The old code seems clearer to me. Could you just change the `iter().cloned()` into `into_iter` and otherwise keep the for loop?

Reply:
I changed it for performance reasons. The for loop uses explicit indices, and this makes Rust do a bound check in every iteration of the loop.

Reply:
I can't see any explicit index usage that should cause a bounds check. Only [] like usage causes one.

Reply:
Hmm, you're right, I must've confused this with another function. I will revert this change, since it doesn't seem to affect performance much.